### PR TITLE
refactor(schemautil): unclutter mutation logic

### PR DESCRIPTION
**Pull request by Serpentiel**
## About this change—what it does

refactors `schemautil` package to unclutter the mutation logic

## Why this way

currently, the mutation logic is too cluttered and complex

this PR aims to reduce the complexity and unclutter the code

### CODE REVIEW

**Comment by Serpentiel** [comment]
```log
=== RUN   TestAccAiven_grafana_user_config
=== PAUSE TestAccAiven_grafana_user_config
=== CONT  TestAccAiven_grafana_user_config
--- PASS: TestAccAiven_grafana_user_config (404.28s)
PASS
```

File: internal/schemautil/service.go [hunk 1]
```diff
@@ -636,19 +636,28 @@ func copyServicePropertiesFromAPIResponseToTerraform(
...
 	}
 
-	NormalizeIpFilter(oldUserConfig, newUserConfig)
+	// Apply in-place user config mutations.
+	if len(oldUserConfig)*len(newUserConfig) != 0 {
+		oldUserConfigFirst := oldUserConfig[0]
+
+		newUserConfigFirst := newUserConfig[0]
+
+		// TODO: This should be implemented in the schema returned by /service_types endpoint instead.
```
Comment (ivan-savciuc) [inline internal/schemautil/service.go:655]: Not sure if we need a TODO for another project here

File: internal/schemautil/mutations.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	newIpFilters, ok := new["ip_filter"].([]interface{})
+	if !ok {
+		return
+	}
+
+	var normalizedIpFilters []interface{}
+
+	// TODO: ipFilterDiff allows for ip_filter field to be managed via Aiven Console aka Web UI.
```
Comment (ivan-savciuc) [inline internal/schemautil/mutations.go:35]: Can you please explain manning of this TODO, and how this is related to TF project?

**Review by ivan-savciuc — COMMENTED** [review]
LGTM, though I need some clarification with the TODOs in the new code, are these TODOs related to Terraform provider or other parts of the Aiven system? 

@muralibasani, please check this out since it touches the code you've recently changed.

File: internal/schemautil/service.go (hunk 1 above)
Comment (Serpentiel) [reply internal/schemautil/service.go:655]: this is a TODO reminder that is supposed to remind us every time when we work with this code that this is not the right way we do things

File: internal/schemautil/mutations.go (hunk 2 above)
Comment (Serpentiel) [reply internal/schemautil/mutations.go:35]: the sorting logic that stores `ip_filter` entries in `ipFilterDiff` is only needed if we want to allow editing them via Aiven Console, or other API clients

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**